### ISSUE TITLE
How to use Docker image in GitLab CI

### ISSUE THREAD

**Issue by djjudas21** [RedCoolBeans/dockerlint#41]
### Issue details

I'm new to Docker and GitLab CI, and I want to use your `redcoolbeans/dockerlint` image in GitLab CI to test my Dockerfile before trying to rebuild the image. However running the `redcoolbeans/dockerlint` image results in an error.

### Steps to reproduce/test case

I have a GitLab repo which contains a `Dockerfile` for an image I want to build, and a `.gitlab-ci.yml` config file for the CI tests.

```yml
# .gitlab-ci-yml
variables:
  IMAGE_TAG: $CI_REGISTRY_IMAGE:$CI_COMMIT_REF_NAME

stages:
  - check
  - build

dockerlint:
  image: redcoolbeans/dockerlint
  stage: check
  script:
    - ""
  only:
    - branches

build:
  image: docker:latest
  services:
    - docker:dind
  stage: build
  script:
    - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN $CI_REGISTRY
    - docker build -t $IMAGE_TAG .
    - docker push $IMAGE_TAG
  tags:
    - dind
```

As I understand it, `redcoolbeans/dockerlint` has an `ENTRYPOINT` which makes it execute `dockerlint` on a `Dockerfile` in the current directory, so I shouldn't need to specify a `script` directive in my CI config. However, `script` is a mandatory parameter so I've set it to a blank script.

The output of my CI test fails with:

```
Running with gitlab-ci-multi-runner 9.5.1 (96b34cc)
  on dind (a5785152)
Using Docker executor with image redcoolbeans/dockerlint ...
Using docker image sha256:f8e4bcf6d77d0ed4db3fe044ad5cbc48d147d388e392989df13626b710479ad7 for predefined container...
Pulling docker image redcoolbeans/dockerlint ...
Using docker image redcoolbeans/dockerlint ID=sha256:ef2fc2ba2a1ce83fefd113345ecc551b98ab04c326208b92db060177b5e9e0f0 for build container...
Running on runner-a5785152-project-257-concurrent-0 via gitlab-ci-dv0.isys.bris.ac.uk...
Fetching changes...
HEAD is now at 8b2334c Try setting a blank script
From https://git.services.bristol.ac.uk/resnet/netops-ci
   8b2334c..ff53251  ci         -> origin/ci
Checking out ff532510 as ci...
Skipping Git submodules setup
ERROR: Cannot open sh.
ERROR: Cannot open sh.
ERROR: Job failed: exit code 1
```

It looks like GitLab CI is trying to use `/bin/sh` as the shell and failing. I tried overriding the entrypoint of your image to `/bin/sh` and then making the script manually execute `dockerlint Dockerfile` but this failed with the same error.

Is the cause of the problem that the image doesn't contain a shell? Any ideas how I can work around this with GitLab CI? I couldn't find any examples online of people using this image with GitLab CI. Hopefully it's a simple fix but I'm just too inexperienced with Docker.

### Affected version(s)

Latest docker image:
```
f8e4bcf6d77d0ed4db3fe044ad5cbc48d147d388e392989df13626b710479ad7
```

**Comment by hobti01** [comment]
We use something like this:
```
dockerfile:
  dependencies: []
  image:
    entrypoint: ["/bin/sh", "-c"]
    name: redcoolbeans/dockerlint
  script: dockerlint -f Dockerfile
```

**Comment by Gwerlas** [comment]
The `script` instruction below should work :

```yaml
dockerlint:
  image: redcoolbeans/dockerlint
  stage: check
  script:
    - "-f Dockerfile"
  only:
    - branches
```

**Comment by QuentinN42** [comment]
My pipeline is currently working with junit reports :
```yaml
.docker-integrity:
  stage: check

  image: projectatomic/dockerfile-lint

  artifacts:
    when: always
    reports:
      junit: report.xml

  before_script:
    - cd "$TEST_PATH"

  script:
    - dockerfile_lint -f Dockerfile -p -u > report.xml

  after_script:
    - mv $TEST_PATH/report.xml $CI_PROJECT_DIR

test-recipe-docker:
  extends: .docker-integrity
  variables:
    TEST_PATH: path1


test-recipe-docker-test:
  extends: .docker-integrity
  variables:
    TEST_PATH: path2

...
```
